### DISCUSSION TITLE
"Unable to serialize object as a key" when running WordPress Starter

### DISCUSSION THREAD

**Discussion by dheesen** [gatsbyjs/gatsby#35457] [Help] [answered]
Hi everyone!

I have a question. When running the WordPress Starter (https://github.com/gatsbyjs/gatsby-starter-wordpress-homepage) I'm getting this error when I run the development command. Checked all the documentation there is on 'gatsbyImageData', but can't the solution to fix this.

 ```
ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Unable to serialize object as a key

  62 | }
  63 |
  64 | fragment HomepageLogoListContent on HomepageLogoList {
  65 |   id
  66 |   text
  67 |   logos {
  68 |     id
  69 |     alt
  70 |     image {
  71 |       id
> 72 |       gatsbyImageData
     |       ^
  73 |       alt
  74 |     }
  75 |   }
  76 | }
  77 |
  78 | fragment HomepageTestimonialListContent on HomepageTestimonialList {
  79 |   id
  80 |   kicker
  81 |   heading
  82 |   content {

File path: /Users/xx/PhpstormProjects/gatsby/xx-fe/src/pages/index.js
Url path: /
Plugin: none



  Error: Unable to serialize object as a key
  
  - index.cjs:114 Object.writeKey
    [xx-fe]/[ordered-binary]/dist/index.cjs:114:11
  
  - read.js:32 Object.getBinaryFast
    [xx-fe]/[lmdb]/read.js:32:46
  
  - read.js:120 Object.get
    [xx-fe]/[lmdb]/read.js:120:22
  
  - caching.js:39 Object.get
    [xx-fe]/[lmdb]/caching.js:39:17
  
  - lmdb-datastore.ts:177 Object.getNode
    [xx-fe]/[gatsby]/src/datastore/lmdb/lmdb-datastore.ts:177:16
  
  - index.ts:18 getNode
    [xx-fe]/[gatsby]/src/datastore/index.ts:18:18
  
  - node-model.js:701 getNodeById
    [xx-fe]/[gatsby]/src/schema/node-model.js:701:41
  
  - node-model.js:142 LocalNodeModel.getNodeById
    [xx-fe]/[gatsby]/src/schema/node-model.js:142:18
  
  - node-model.js:632 ContextualNodeModel.getNodeById
    [xx-fe]/[gatsby]/src/schema/node-model.js:632:27
  
  - gatsby-node.js:12 resolver
    /Users/xx/PhpstormProjects/gatsby/xx-fe/gatsby-node.js:12:42
  
  - resolvers.ts:683 wrappedTracingResolver
    [xx-fe]/[gatsby]/src/schema/resolvers.ts:683:20
```

**Comment by robmarshall** [comment]
I am seeing this as well on one of my clients sites. But only seems to happen on a content change build. Cannot work it out!

edit: typo

**Comment by chris-rohrer** [comment]
I just ran into the same issue.

Running 'gatsby clean' and then 'gatsby develop' fixed it for me. 

By running gatsby clean it deletes the node modules cache which seems to have been the issue. Without the cache, it then performs gatsby-plugin-sharp.IMAGE_PROCESSING jobs, which haven't been run before deleting the cache. Or, at least that's my theory.

**Comment by aghreed** [comment, marked as answer]
With respect to the WordPress Homepage Starter, the issue is in the `gatsby-node.js` file here in the `wpImagePassthroughResolver`: https://github.com/gatsbyjs/gatsby-starter-wordpress-homepage/blob/main/gatsby-node.js#L13

Updating from `id: source.localFile` -> `id: source.localFile.id` fixes the issue.

I'll be updating the code in the WordPress Homepage Starter here shortly, but wanted to share here first.
